docs/fixture_usage.md

```diff
@@ -0,0 +1,119 @@
+# Guidelines for fixtures usage in OCS-CI
+
+## Basic information
+
+What is pytest Fixture? You can find definition and examples documentation in
+[pytest web](https://docs.pytest.org/en/latest/fixture.html). Please read this
+as the first.
+
+We should keep our fixtures organized and have all the dependencies well
+defined in fixtures.
+
+All the fixtures should be documented - mandatory is document return value.
+
+Fixture is compose of its setup part, the code which is done for prepare
+resource and teardown part which is after yield of added via
+`request.addfinalizer` method. It's strictly forced to use `addfinalizer`
+method cause it can happen that what is after yield won't be proceed when
+exception will be thrown in the middle of the setup.
+
+The finalizer should be added as soon as possible to make sure the finalizer
+will be called at the end. (Otherwise we will hit the same issue as we have
+with yield)
+
+## Use cases
+
+TODO: Mention function factories and use cases and when to use which approach.
+Something like:
+
+* Setup resource,  test (using resources) , teardown - regular usage of fixture
+* Test creating resources as part of test, teardown of resources - fixture
+  factory solve the issue here.
+* Setup resources,  test (using resources , deleting resources), teardown
+  shouldn’t fail on those deleted resources
+
+## Avoid
+
+Here is the list of things we should avoid in our fixtures:
+
+* Using references to `request.node.cls` for set, altering or reading class
+  attributes.
+* Using globals for sharing data.
+* Using finalizer after yield.
```

Comment:
We won't be using `yield` at all if we are enforcing the use of `request.addfinalizer()`.

Reply:
You can in some cases use fixture with yield if you do not need to do any cleanup/teardown on maybe.  That's why I wrote it like that.

Reply:
If there isn't any cleanup/teardown to perform we would use `return` and not `yield`. I think any scenario in which we would need yield would mean we need to add a finalizer instead.

Reply:
I agree with @clacroix12 here, I don't really see a strong reason to use ``yield`` at all.

Reply:
ok, removing in next commit